### ISSUE TITLE
Can't query nested array

### ISSUE THREAD

**Issue by rptony** [OpenKitten/MongoKitten#47]
I have a document of the following format.

```
{
    "_id" : ObjectId("17e2fb57247d5ef8cd6f4447"),
    "users" : [
        {
            "Name" : "Brad",
            "Age" : 42
        },
        {
            "Name" : "Paul",
            "Age" : 32
        }
    ]
}

```

There is no way to query the embedded array with `Query` builder. How would you `find` the record `"Name" == "Brad"`

**Comment by Joannis** [comment]
For this I'd use a raw Document rather than the query builder. The query builder isn't sophisticated enough for more complex queries just yet.

However a raw Document gets to use all of MongoDB's features like searching through text indexes, and looking through arrays.

So what I'd do is use [elemMatch](https://docs.mongodb.com/v2.6/reference/operator/query/elemMatch/) to search through the Array elements for an element containing a Document with the "Name" of "Brad".

```swift
let query: Document = [
  "users": [
    "elemMatch": ["Name": "Brad"]
  ]
]
```

It's not been tested by me, but I think this should help.

I'll ensure that these kinds of more complex queries are easier to build in the next version.

**Comment by rptony** [comment]
That returned the entire  document. (Had to stuff a `$` prior to `elemeMatch`)
```
 let query: Document = [
                "users": [
                    "$elemMatch": ["Name": "Brad"]
                ]
            ]
            
  let answer = try! collection.findOne(matching: query)
```

**Comment by Joannis** [comment]
Ah.. I didn't know you only wanted to fetch the subdocument.

**Comment by gabriellanata** [comment]
This issue is also solved by merge request #77 which adds support for ElemMatch in AQT

**Comment by Joannis** [comment]
Good one! I'll close this
